boa/src/syntax/ast/node/iteration.rs

```diff
@@ -137,3 +137,175 @@ impl InnerForLoop {
+        self.expr().display(f, indentation)
+    }
+}
+
+impl fmt::Display for WhileLoop {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.display(f, 0)
+    }
+}
+
+impl From<WhileLoop> for Node {
+    fn from(while_loop: WhileLoop) -> Self {
+        Self::WhileLoop(while_loop)
+    }
+}
+
+/// The `do...while` statement creates a loop that executes a specified statement until the
+/// test condition evaluates to false.
+///
+/// The condition is evaluated after executing the statement, resulting in the specified
+/// statement executing at least once.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///  - [MDN documentation][mdn]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-do-while-statement
+/// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/do...while
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Clone, Debug, Trace, Finalize, PartialEq)]
+pub struct DoWhileLoop {
+    body: Box<Node>,
+    cond: Box<Node>,
+}
+
+impl DoWhileLoop {
+    pub fn body(&self) -> &Node {
+        &self.body
+    }
+
+    pub fn cond(&self) -> &Node {
+        &self.cond
+    }
+
+    /// Creates a `DoWhileLoop` AST node.
+    pub fn new<B, C>(body: B, condition: C) -> Self
+    where
+        B: Into<Node>,
+        C: Into<Node>,
+    {
+        Self {
+            body: Box::new(body.into()),
+            cond: Box::new(condition.into()),
+        }
+    }
+
+    pub(super) fn display(&self, f: &mut fmt::Formatter<'_>, indentation: usize) -> fmt::Result {
+        write!(f, "do")?;
+        self.body().display(f, indentation)?;
+        write!(f, "while ({})", self.cond())
+    }
+}
+
+impl fmt::Display for DoWhileLoop {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.display(f, 0)
+    }
+}
+
+impl From<DoWhileLoop> for Node {
+    fn from(do_while: DoWhileLoop) -> Self {
+        Self::DoWhileLoop(do_while)
+    }
+}
+
+/// The `continue` statement terminates execution of the statements in the current iteration of
+/// the current or labeled loop, and continues execution of the loop with the next iteration.
+///
+/// The continue statement can include an optional label that allows the program to jump to the
+/// next iteration of a labeled loop statement instead of the current loop. In this case, the
+/// continue statement needs to be nested within this labeled statement.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///  - [MDN documentation][mdn]
+///
+/// [spec]: https://tc39.es/ecma262/#prod-ContinueStatement
+/// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/continue
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Clone, Debug, Trace, Finalize, PartialEq)]
+pub struct Continue {
+    label: Option<Box<str>>,
+}
+
+impl Continue {
+    pub fn label(&self) -> &Option<Box<str>> {
+        &self.label
+    }
```

Comment:
This should also return an `Option<&str>`:
```suggestion
    pub fn label(&self) -> Option<&str> {
        self.label.as_ref().map(Box::as_ref)
    }
```